cluster-api-gcp/CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+# Contributing Guidelines
+
+## Build
+
+```bash
+$ cd $GOPATH/src/k8s.io/
+$ git clone git@github.com:kubernetes/kube-deploy.git
+$ cd kube-deploy/cluster-api-gcp/
+$ go build
+```
+
+This will create a binary `cluster-api-gcp` that you can use to manage a GCP cluster.
+
+## Developing
+
+When making changes to the machine controller, it's generally a good idea to delete any existing cluster created with an older version of the cluster-api.
+
+```bash
+$ ./cluster-api-gcp delete
+```
+
+After making changes to the machine controller or the actuator, you need to follow these two steps:
+
+1. Rebuild the machine-controller image.
```

Comment:
Not everyone has (or should have) permissions to upload images to the official repository. We may need to think of some alternate path for developers to do development without polluting the official images.

Reply:
https://github.com/kubernetes/kube-deploy/issues/516